src/schema.c

```diff
@@ -186,7 +263,9 @@ ldap_schema_append_attributetype(struct ldap_schema_t *schema, LDAPAttributeType
 bool
 ldap_schema_append_objectclass(struct ldap_schema_t *schema, LDAPObjectClass *objectclass)
 {
-    if (!schema || !objectclass)
+    char *objectclass_name = ldap_objectclass2name(objectclass);
```

Comment:
By using `ldap_objectclass2name` you only select first name from a list of attributetype names. Consider addition of all names.

- https://git.openldap.org/openldap/openldap/-/blob/master/libraries/libldap/schema.c#L36
- https://git.openldap.org/openldap/openldap/-/blob/master/libraries/libldap/schema.c#L70